tests/scripts/all.sh

```diff
@@ -873,7 +873,12 @@ component_check_doxygen_warnings () {
     tests/scripts/doxygen.sh
 }
 
+component_check_test_requires_psa_disabled () {
+  msg "Check: tests requiring PSA to be disabled"
 
+  not grep -n -R 'depends.*!MBEDTLS_USE_PSA_CRYPTO' tests/suites/
+  not grep -n -R 'requires.*disabled.*USE_PSA' tests/ssl-opt.sh tests/opt-testcases/
```

Comment:
Please don't use `grep -R`: it can search spurious files in developers' working directories (e.g. backups). List the expected directories and file name patterns (typically extensions).

Also please avoid overly general regexps that might have spurious matches.

```suggestion
  not grep -n 'depends_on:.*!MBEDTLS_USE_PSA_CRYPTO' tests/suites/*.function tests/suites/*.data
  not grep -n '^ *requires_config_disabled.*MBEDTLS_USE_PSA_CRYPTO' tests/ssl-opt.sh tests/opt-testcases/*.sh
```